### PR TITLE
Amazon + Demo (ROQ code + deployment)  + local deployment

### PR DESCRIPTION
**Pull request by vanmellebenjamin**
This huge pull request contains more than one feature (that will be note the case for the next pull requests).

Dest:
@jbruggem
@ahderman 
@sskhiri 

This pull request contains:
- A readme which explains how to use the scripts for each use case.
- The Amazon deployment script (Ansible), roq-deployment/amazon/*
- The ROQ code for demonstration: roq-demonstration/*
- The deployment code for demonstration (Ansible + Docker): roq-deployment/demonstration/\* &   roq-deployment/docker/\* & roq-deployment/ansible/*
- The deployment code for local installation (Ansible): roq-deployment/development/\* & roq-deployment/ansible/*
- Small code corrections ../MngmtController.java 

Feel free to ask me and to give me feedbacks !

### CODE REVIEW

**Comment by sskhiri** [comment]
@ahderman can you please look at the Travis CI and define why the build fails with that branch ? Thanks!

**Comment by ahderman** [comment]
The Travis CI build fails because we have not finished setting it up. It is not a problem of the code in this pull request.

File: roq-deployment/amazon/GCM/templates/roqgcm.j2 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+		PID=$!
+
+		log_begin_msg "GCM Started [$PID]"
+
+		echo $PID > $GCM_PIDFILE
+		;;
+
+	stop)
+        fail_unless_root
```
Comment (ahderman) [inline roq-deployment/amazon/GCM/templates/roqgcm.j2:40]: Why do you check twice if you are root? Once on line 24 and once here.

File: roq-deployment/amazon/GCM/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+- name: Creates .roq directory
+  file: path=~/.roq state=directory mode=777
```
Comment (ahderman) [inline roq-deployment/amazon/GCM/tasks/main.yml:2]: The ~/.roq folder could probably be a variable, since you use it in several places.

File: roq-deployment/amazon/HCM/templates/roqhcm.j2 [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		sleep 10
+		$0 start
+		;;
+	*)
+		echo "Usage: $0 {start|stop|status|restart}"
+		exit 1
+		;;
+esac
+
+exit 0
```
Comment (ahderman) [inline roq-deployment/amazon/HCM/templates/roqhcm.j2:74]: This file looks very similar to the GCM one. Is there no way of merging them?

File: roq-deployment/amazon/getROQ/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
+    JAVA_HOME: /usr/lib/jvm/java-7-openjdk-i386
+  when: ansible_architecture == "i386"
+
+- name: install ROQ
+  shell: "cd {{roq_path}}; mvn install -Dmaven.test.skip=true"
+  environment:
+    JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
+    JAVA_HOME: /usr/lib/jvm/java-7-openjdk-amd64
+  when: ansible_architecture == "x86_64"
```
Comment (ahderman) [inline roq-deployment/amazon/getROQ/tasks/main.yml:19]: Same here: we repeat an action when we could probably use `register` for defining the JAVA_HOME.

File: roq-deployment/ansible/gimme-roq/files/pomAlt.xml [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+    </dependencies>
+    <build>
+        <sourceDirectory>src</sourceDirectory>
+        <testSourceDirectory>test/src</testSourceDirectory>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-gpg-plugin</artifactId>
+<configuration> <skip>true</skip> </configuration>
+                <configuration> <skip>true</skip> </configuration>
```
Comment (ahderman) [inline roq-deployment/ansible/gimme-roq/files/pomAlt.xml:166]: Why is this file here? There are two similar pom files in this pull request, so one of them is probably redundant. I suppose it is this one. Lines 165 and 166 are the same.

File: roq-deployment/ansible/gimme-roq/handlers/main.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+- name: Restart zookeeper
+  service: name=zookeeper state=restarted enabled=yes
+
+- name: Restart zookeeper fedora
+  shell: "/bin/zookeeper/zookeeper/bin/zkServer.sh restart"
```
Comment (ahderman) [inline roq-deployment/ansible/gimme-roq/handlers/main.yml:5]: Do you need a conditional `where` here?

File: roq-deployment/ansible/gimme-roq/templates/zoo.cfg.j2 [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+{% for host in groups.test %}
+server.{{hostvars[host].zookeeper_id }}={{hostvars[host].ipv4_address|default(hostvars[host]['ansible_default_ipv4']['address']) }}:2888:3888
+{% endfor %}
+
+{% else %}
+
+server.1=localhost:2888:3888
+
+{% endif %}
```
Comment (ahderman) [inline roq-deployment/ansible/gimme-roq/templates/zoo.cfg.j2:25]: Some files (including this one) are duplicated in the /ansible and /amazon folders.
- why is the /amazon folder not inside /ansible?
- is there no other way than to keep two copies of the same files?

**Comment by vanmellebenjamin** [comment]
Thank you @ahderman, I'll bring the corrections next week.

File: roq-deployment/ansible/gimme-roq/handlers/main.yml (hunk 6 above)
Comment (vanmellebenjamin) [inline roq-deployment/ansible/gimme-roq/handlers/main.yml:5]: The handler is useless, I'll remove it

File: roq-deployment/ansible/gimme-roq/files/pomAlt.xml (hunk 5 above)
Comment (vanmellebenjamin) [inline roq-deployment/ansible/gimme-roq/files/pomAlt.xml:166]: This file had been used for test purpose, this is a duplication, I'll remove it

File: roq-deployment/amazon/getROQ/tasks/main.yml (hunk 4 above)
Comment (vanmellebenjamin) [inline roq-deployment/amazon/getROQ/tasks/main.yml:19]: Right, I thonk about that, but I didn't found quickly enough the way to do it.

File: roq-deployment/amazon/HCM/templates/roqhcm.j2 (hunk 3 above)
Comment (vanmellebenjamin) [inline roq-deployment/amazon/HCM/templates/roqhcm.j2:74]: This is implemented as two different services, so I don't think that it is pertinent to merge the files

File: roq-deployment/amazon/GCM/tasks/main.yml (hunk 2 above)
Comment (vanmellebenjamin) [inline roq-deployment/amazon/GCM/tasks/main.yml:2]: Yes we could

File: roq-deployment/amazon/GCM/templates/roqgcm.j2 (hunk 1 above)
Comment (vanmellebenjamin) [inline roq-deployment/amazon/GCM/templates/roqgcm.j2:40]: mistake :)

File: roq-deployment/ansible/gimme-roq/templates/zoo.cfg.j2 (hunk 7 above)
Comment (vanmellebenjamin) [inline roq-deployment/ansible/gimme-roq/templates/zoo.cfg.j2:25]: Haha I waited for this remark, actually I could and I should. I will study next week this possibility

**Comment by ahderman** [comment]
The corrections to this pull request can be introduced in a later commit. There are no serious issues anyway.
